Tasks/VsTest/vstest.ts

```diff
@@ -84,34 +84,49 @@ catch (error) {
     throw error;
 }
 
+function getResolvedPattern(pattern: string): string {
+    if (path.isAbsolute(pattern)) {
+        return pattern;
+    }
+    else {
+        return path.join(sourcesDirectory, pattern);
+    }
+}
+
+function getPatternWithoutIncludeExclude(pattern: string): string {
+    return pattern.startsWith('-:') || pattern.startsWith('+:') ? pattern.substr(2) : pattern;
+}
+
 function getTestAssemblies(): Set<string> {
-    var testAssemblyFiles = [];
+    let testAssemblyFiles: Array<string> = [];
     if (testAssembly.indexOf('*') >= 0 || testAssembly.indexOf('?') >= 0) {
         tl.debug('Pattern found in solution parameter.');
-        var excludeTestAssemblies = [];
-        var allFiles = tl.find(sourcesDirectory);
-        var testAssemblyFilters = testAssembly.split(';');
+        let excludeTestAssemblies: Array<string> = [];
+        let allFiles: Array<string> = tl.find(sourcesDirectory);
+        let testAssemblyFilters: Array<string> = testAssembly.split(';');
         testAssemblyFilters.forEach(function (testAssemblyFilter) {
-            if (testAssemblyFilter.startsWith("-:")) {
-                if (testAssemblyFilter.indexOf('*') >= 0 || testAssemblyFilter.indexOf('?') >= 0) {
-                    excludeTestAssemblies = excludeTestAssemblies.concat(getFilteredFiles(testAssemblyFilter.substr(2), allFiles));
-                }
-                else {
-                    excludeTestAssemblies.push(testAssemblyFilter.substr(2));
-                }
-            }
-            else if (testAssemblyFilter.indexOf('*') >= 0 || testAssemblyFilter.indexOf('?') >= 0) {
-                testAssemblyFiles = testAssemblyFiles.concat(getFilteredFiles(testAssemblyFilter, allFiles));
+            let exclude: boolean = testAssemblyFilter.startsWith('-:');
+            let hasQuantifier: boolean = Math.max(testAssembly.indexOf('*'), testAssembly.indexOf('?')) != -1;
+            let patternWithoutIncludeExclude: string = getPatternWithoutIncludeExclude(testAssemblyFilter);
```

Comment:
ts will auto-resolve to `string` if it's strongly typed call. not required to `strong` name the variable everytime